File: app/src/main/java/com/gb/rating/models/repository/VerifiedRatingRepository.java
package com.gb.rating.models.repository;

import com.gb.rating.models.VerifiedRating;
import java.util.List;
import io.reactivex.Maybe;
import io.reactivex.annotations.NonNull;

public interface VerifiedRatingRepository {

    @NonNull io.reactivex.Completable writeVerifiedRating(@NonNull VerifiedRating value, @NonNull String iMEI);
    @NonNull Maybe<List<VerifiedRating>> retrieveVerifiedRatingList(@NonNull String iMEI);
}

File: app/src/main/java/com/gb/rating/models/repository/VerifiedRatingRepository.kt
package com.gb.rating.models.repository

import com.gb.rating.models.VerifiedRating
import io.reactivex.Maybe
import io.reactivex.annotations.NonNull

interface VerifiedRatingRepository {

    @NonNull
    fun writeVerifiedRating(@NonNull value: VerifiedRating, @NonNull iMEI: String): io.reactivex.Completable

    @NonNull
    fun retrieveVerifiedRatingList(@NonNull iMEI: String): Maybe<List<VerifiedRating>>
}

